apps/expo-go/android/expoview/src/main/java/host/exp/exponent/kernel/ExponentPackageLogger.kt

```diff
@@ -0,0 +1,118 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+package host.exp.exponent.kernel
+
+import android.net.Uri
+import android.util.Log
+import androidx.core.net.toUri
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.WebSocket
+import okhttp3.WebSocketListener
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
+
+class ExponentPackageLogger private constructor(private val appUrl: Uri) {
+  enum class LogLevel {
+    WARNING,
+    ERROR,
+    INFO
+  }
+
+  private var okHttpClient: OkHttpClient = OkHttpClient.Builder()
+    .connectTimeout(10, TimeUnit.SECONDS)
+    .writeTimeout(10, TimeUnit.SECONDS)
+    .readTimeout(0, TimeUnit.MINUTES) // Disable timeouts for read
+    .build()
+
+  private var webSocket: WebSocket? = null
+  private var connected: Boolean = false
+  private var pendingMessage: String? = null
+  private var logLevel: LogLevel = LogLevel.INFO
+
+  private fun sendMessage(message: String, logLevel: LogLevel): CompletableFuture<Void> {
```

Comment:
use kotlin coroutine than the java future

Reply:
Fixed